src/keywords/rFactorRangesVector.cpp

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "keywords/rFactorRangesVector.h"
+#include "base/lineParser.h"
+#include "classes/coreData.h"
+
+RFactorRangesVector::RFactorRangesVector(std::vector<Range> &data) : KeywordBase(typeid(this)), data_(data) {}
+
+/*
+ * Data
+ */
+
+// Return the data vector
+std::vector<Range> &RFactorRangesVector::data() { return data_; }
+const std::vector<Range> &RFactorRangesVector::data() const { return data_; }
+
+/*
+ * Arguments
+ */
+
+// Deserialise from supplied LineParser, starting at given argument offset
+bool RFactorRangesVector::deserialise(LineParser &parser, int startArg, const CoreData &coreData)
+{
+    // Loop over arguments
+    for (auto n = startArg; n < parser.nArgs(); n += 2)
+    {
+
+        // Check to see if value pair are numbers
+        if (DissolveSys::isNumber(parser.argsv(n)))
+        {
+            if (DissolveSys::isNumber(parser.argsv(n + 1)))
+            {
+                // If both are numbers, initialise new Range
+                data_.emplace_back(parser.argd(n), parser.argd(n + 1));
+            }
+            else
+            {
+                return Messenger::error("Value '{}' provided for keyword R Factor Ranges doesn't appear to be a number.\n",
+                                        parser.argsv(n + 1));
+            }
+        }
+        else
+        {
+            return Messenger::error("Value '{}' provided for keyword R Factor Ranges doesn't appear to be a number.\n",
+                                    parser.argsv(n));
```

Comment:
Slight adjustment to the message, and include the keyword `name()` for the user's benefit:
```suggestion
            return Messenger::error("Value '{}' provided to keyword '{}' doesn't appear to be a number.\n",
                                    parser.argsv(n), name_);
```